api/views/personal_info.py

```diff
@@ -19,8 +21,40 @@ def get_serializer_class(self) -> type[BaseSerializer[PersonalInfo]]:
     @action(detail=False, methods=["post"])
     def check_existence(self, request: Request) -> Response:
         """
-        Check if a personal info exists.
+        Checks if a personal info exists.
         """
         serializer = self.get_serializer(data=request.data)
         serializer.is_valid(raise_exception=True)
         return Response(status.HTTP_200_OK)
+
+    @extend_schema(
+        description=(
+            "Checks if personal info exists with this chat ID of Telegram registration bot."
+        ),
+        parameters=[
+            OpenApiParameter(name="chat_id", type=str, required=True),
+        ],
+        responses={
+            200: OpenApiResponse(description="Chat ID is found"),
+            404: OpenApiResponse(description="Chat ID not found"),
+        },
+    )
+    @action(detail=False, methods=["get"])
+    def check_existence_of_chat_id(self, request: Request) -> Response:
+        """
+        Check if PersonalInfo with given ``registration_telegram_bot_chat_id`` exists.
+        """
+        # This method is used without a serializer because this seems to be overkill
+        # for a simple GET request with a single parameter.  Since name and e-mail are required
+        # fields for PersonalInfo (and we only have chat ID), we cannot use the same technique
+        # as in .check_existence(). Note that .check_existence() above returns 200 OK if the
+        # personal info does NOT exist and 400 BAD REQUEST if it DOES exist.
+        # This is because POST method is used up there, so non-existent user can be "created",
+        # hence success is reported if no user was found.
+        # Here we use GET and the responses are different.
+
+        if PersonalInfo.objects.filter(
+            registration_telegram_bot_chat_id=request.GET.get("chat_id")
```

Comment:
If I will call this method without "chat_id" param - it will return 200 if we have PersonalInfo with  registration_telegram_bot_chat_id=None.

I think we should make this param required, and then it's better to put this checking logic into serializer

Reply:
Well actually it makes absolutely no sense to call this view without chat ID (a bot will always have some ID), but you're right in that we should make an implementation that doesn't care about who is using it.  

Maybe I don't understand serializers, but how can we use it if we're in the very beginning and haven't even tried to create a `PersonalInfo`, we have no name, no email yet (and they are required for `PersonalInfo`).

For now, I just added the logic for handling the missing `chat_id`. Maybe that will do :)

Reply:
Serializer is not required to be ModelSerializer. It could be just description of some fields with `validate` method. And here in view you should just pass request.query_params into it, as data.

Reply:
I mean just regular Serializer class.

https://www.django-rest-framework.org/tutorial/1-serialization/
https://www.django-rest-framework.org/api-guide/serializers/

Reply:
Done